src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,610 @@
+        [CLSCompliant(false)]
+        public bool TryGetUInt16(out ushort value) => ushort.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its UInt32 equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the UInt32 value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  // <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        [CLSCompliant(false)]
+        public bool TryGetUInt32(out uint value) => uint.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its UInt64 equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the UInt64 value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  // <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        [CLSCompliant(false)]
+        public bool TryGetUInt64(out ulong value) => ulong.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its Decimal equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the Decimal value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  // <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        public bool TryGetDecimal(out decimal value) => decimal.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified value.
+        /// </summary>
+        /// <param name="value">The string representation of a numeric value in a legal JSON number format.</param>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided value is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   Provided value is not in a legal JSON number format.
+        /// </exception>
+        public void SetFormattedValue(string value)
+        {
+            if (value == null)
```

Comment:
@ahsonkhan question: is System.Text.Json going to be annotated with nullability?

Reply:
> is System.Text.Json going to be annotated with nullability?

Eventually, yes, along with the rest of corefx.  For 3.0 we've only annotated a subset of the reference assemblies and corresponding implementations, namely those that have implementations in corelib.